pkg/assembler/helpers/purl.go

```diff
@@ -0,0 +1,141 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helpers
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	purl "github.com/package-url/packageurl-go"
+
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+)
+
+const PurlTypeGuac = "guac"
```

Comment:
should we have another helper function that takes in a string and return a generic `pkg:guac` purl. This "guac purl" can be passed in purlConvert to define the `model.Package`

Reply:
oo yea this would be a good one, but i think it lands inside the parser instead of assembler

Reply:
Should we add a definition here to parse `pkg:guac`?